src/data/list/basic.lean

```diff
@@ -2975,6 +2978,40 @@ instance decidable_suffix [decidable_eq α] : ∀ (l₁ l₂ : list α), decidab
   | s, t, ⟨b::l, he⟩ := list.no_confusion he (λab lt, or.inr ⟨l, lt⟩)
   end⟩
 
+lemma inits_cons (a : α) (l : list α) : inits (a :: l) = [] :: l.inits.map (λ t, a :: t) :=
+by simp
+
+lemma tails_cons (a : α) (l : list α) : tails (a :: l) = (a :: l) :: l.tails :=
+by simp
+
+@[simp]
```

Comment:
Is the RHS really simpler than the LHS? Similar for the lemma below this one.

Reply:
I don't feel strongly. I figured if `*_cons` is `simp`, the inductive nature of `++` means that `*_append` should be too. Is there an existing counterexample to that claim?

Reply:
They seem useful enough to be `simp` lemmas. I think the following looks nicer and matches `tails_append` better though.
```lean
lemma inits_append : ∀ (s t : list α), inits (s ++ t) = s.inits.init ++ t.inits.map (λ l, s ++ l)
```

Reply:
I was not aware of `.init`, but I agree the symmetry is nice

Reply:
However, `.tail` tends to be easier to work with inductively I think

Reply:
That's true. Let's stick to `tail` then.